tasks/ceph.py

```diff
@@ -71,10 +72,79 @@ def ceph_log(ctx, config):
             )
         )
 
+    class Rotater:
+        stopping = False
+        def invoke_logrotate(self):
+            #1) install ceph-test.conf in /etc/logrotate.d
+            #2) continuously loop over logrotate invocation with ceph-test.conf
+            while not self.stopping:
+                time.sleep(30)
+                run.wait(
+                    ctx.cluster.run(
+                        args=['sudo', 'logrotate', '/etc/logrotate.d/ceph-test.conf'
+                          ],
+                        wait=False,
+                    )
+                )
+
+        def begin(self):
+            self.thread = gevent.spawn(self.invoke_logrotate)
+
+        def end(self):
+            self.stopping = True
+            self.thread.get()
+            
+    def write_rotate_conf(ctx, daemons):
+        testdir = teuthology.get_testdir(ctx)
+        rotate_conf_path = os.path.join(os.path.dirname(__file__), 'logrotate.conf')
+        with file(rotate_conf_path, 'rb') as f:
+            conf = ""
+            for daemon,size in daemons.iteritems():
+                log.info('writing logrotate stanza for {daemon}'.format(daemon=daemon))
+                conf += f.read().format(daemon_type=daemon,max_size=size)
+                f.seek(0, 0)
+            
+            for remote in ctx.cluster.remotes.iterkeys():
+                teuthology.write_file(remote=remote,
+                                      path='{tdir}/logrotate.ceph-test.conf'.format(tdir=testdir),
+                                      data=StringIO(conf)
+                                  )
+                remote.run(
+                    args=[
+                        'sudo',
+                        'mv',
+                        '{tdir}/logrotate.ceph-test.conf'.format(tdir=testdir),
+                        '/etc/logrotate.d/ceph-test.conf',
+                        run.Raw('&&'),
+                        'sudo',
+                        'chmod',
+                        '0644',
+                        '/etc/logrotate.d/ceph-test.conf',
+                        run.Raw('&&'),
+                        'sudo',
+                        'chown',
+                        'root.root',
+                        '/etc/logrotate.d/ceph-test.conf'
+                    ]
+                )
+
+    if ctx.config.get('log-rotate'):
```

Comment:
Mmm, yeah. TBH I just did that because I got frustrated with getting all the nested structures right (*blush*) but I also like it because we might want to rotate non-Ceph stuff in the future. Right now it's part of ceph.py but given the structure we can move it elsewhere pretty easily.